src/main/java/org/spongepowered/common/data/processor/value/item/ItemMapWorldValueProcessor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.value.item;
+
+import com.flowpowered.math.vector.Vector2i;
+import net.minecraft.world.storage.MapData;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.value.ValueContainer;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.bridge.world.WorldServerBridge;
+import org.spongepowered.common.data.processor.common.AbstractSpongeValueProcessor;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+import org.spongepowered.common.world.WorldManager;
+
+import java.util.Optional;
+
+public class ItemMapWorldValueProcessor extends AbstractSpongeValueProcessor<MapData, World, Value<World>> {
+
+    public ItemMapWorldValueProcessor() {
+        super(MapData.class, Keys.MAP_WORLD);
+    }
+
+    @Override
+    protected Value<World> constructValue(World actualValue) {
+
+        return new SpongeValue<>(Keys.MAP_WORLD,
+                Sponge.getServer().getWorld(Sponge.getServer().getDefaultWorldName()).get(), actualValue);
+    }
+
+    @Override
+    protected boolean set(MapData container, World value) {
+        container.dimension = (byte)((WorldServerBridge)value).bridge$getDimensionId();
```

Comment:
This is where things are so much fun.

Mods very much have dimension ids of worlds well above 256. Since they can set the number to whatever they like! Forge patches this to be an int for a good reason..

@gabizou How do you want to deal with this?

Reply:
this should be using the .setDimensionId() to access this field, and since bridge$getDimensionId() shouldn't return a value that the field can't hold - eg. in SpongeVanilla it wont return more than a byte's max value and in SpongeForge the field is able to take it.